packages/common/src/search/upgradeCatalogSchema.ts

```diff
@@ -0,0 +1,71 @@
+import { getProp } from "../objects";
+import { cloneObject } from "../util";
+import { IHubCatalog } from "./types";
+
+const CATALOG_SCHEMA_VERSION = 1.0;
+
+/**
+ * Apply schema upgrades to Catalog objects
+ * @param catalog
+ * @returns
+ */
+export function upgradeCatalogSchema(catalog: any): IHubCatalog {
```

Comment:
Nice idea. Is the plan that we always invoke this as part of the workflow to save a catalog? Or is it a wrapper for when we simply retrieve a catalog and need it to be structured according the latest schema?

Reply:
The latter. The way we currently handle "migrations" for Sites, Initiatives and a few other things in the system is to always run things through these functions "at read". Then the rest of the system can reliably work with the latest schema version structure. If the object happens to be saved, then it's saved with the latest version.